### PR TITLE
 feat: add optional trace to CairoVM struct and add enable trace flag

### LINKED ISSUE #9: feat: add optional trace to CairoVM struct and add enable trace flag
Right now the CairoVM is [missing an optional trace field](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/core.zig#L14), which when we are in `enable-trace` mode should be updated with every step of the virtual machine. The trace is a vector of `TraceEntry` of the registers `pc`, `ap`, and `fp`. In `runInstruction` `[function](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/core.zig#L104)` we need to push a new `TraceEntry` struct that receives the registers from `RunContext` [register holder](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/run_context.zig#L12). 

It's OK to push a default value of `RunContext` for the moment but when [this](https://github.com/keep-starknet-strange/cairo-zig/issues/10) is complete the actual value needs to be pushed. 

Add the `enable-trace` flag in [cmd](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/cmd/cmd.zig).

### PR DESCRIPTION
**Pull request by itzlambda**
fix: #9

### CODE REVIEW

File: src/vm/core.zig [hunk 1]
```diff
@@ -37,17 +41,23 @@ pub const CairoVM = struct {
     /// - `CairoVM`: The created VM.
     /// # Errors
     /// - If a memory allocation fails.
-    pub fn init(allocator: *const Allocator) !CairoVM {
+    pub fn init(allocator: *const Allocator, config: Config) !CairoVM {
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:44]: add the argument in the doc of the function please

File: src/vm/core.zig [hunk 2]
```diff
@@ -37,17 +41,23 @@ pub const CairoVM = struct {
...
         // Initialize the memory segment manager.
         const memory_segment_manager = try segments.MemorySegmentManager.init(allocator);
         // Initialize the run context.
         const run_context = try RunContext.init(allocator);
 
+        var trace: ?ArrayList(TraceEntry) = null;
+        if (config.enable_trace) {
+            trace = try ArrayList(TraceEntry).initCapacity(allocator.*, 4096);
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:52]: can you write a TODO comment with a link to the issue to make the initial capacity configurable please ?

File: src/vm/core.zig [hunk 3]
```diff
@@ -913,3 +941,60 @@ test "update fp dst felt" {
...
+test "trace is enabled" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    // Initialize an allocator.
+    var allocator = std.testing.allocator;
+
+    // Create a new VM instance.
+    var config = Config{ .proof_mode = false, .enable_trace = false };
+    config.enable_trace = true;
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:954]: why dont you put directly the true value when initiating ?

**Review by AbdelStark — CHANGES_REQUESTED** [review]
left some comments

File: src/vm/core.zig (hunk 3 above)
Comment (lambda-0x) [reply src/vm/core.zig:954]: oh initially i was using `Config.default()` there, then used regex to replace it :). will update.

File: src/vm/core.zig (hunk 1 above)
Comment (lambda-0x) [reply src/vm/core.zig:44]: sorry missed that.

**Review by AbdelStark — APPROVED** [review]
lgtm

**Pull request merged**